core/src/scss/utilities/mixins/icons-logos/_icon.scss

```diff
@@ -0,0 +1,26 @@
+@charset "UTF-8";
+
+///
+/// Display an icon using the :before on an element. If needs to be part of a link, consider using the link-icon instead.
+///
+/// @name icon
+///
+/// @param {string} $icon  - Basename of link icon .svg file (without the .svg suffix).
+/// @param {string} $width  - Width of icon including unit, e.g. 0.75em, 24px.
+/// @param {string} $vertical  - Adjustment of vertical position of icon, e.g., '2px' moves icon up 2 pixels, '-0.3em' moves it down 0.3 em.
+///
+/// @group mixin
+@mixin icon($icon, $width: 0.65em, $vertical: null) {
+  text-decoration: none;
+  @supports (mask-repeat: no-repeat) {
+    &::before {
+      @include margin(null 0.3em $vertical 0.4em);
+      @include size($width);
+      display: inline-block;
+      content: '';
+      mask: url("#{$su-image-path}/#{$icon}.svg") no-repeat 0 0;
```

Comment:
It might be better for us to not assume path but to pass in the whole path to the $icon param as it could open this up for others to use their own bucket of icons.